### PR TITLE
Control the Retry policy values to stop the test from being flaky

### PR DESCRIPTION
**Pull request by LukeButters**
# Background

This test has been flaky, since it measures port close time as well.

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Halibut.Tests/PollingClientConnectionHandlingFixture.cs [hunk 1]
```diff
@@ -39,7 +40,7 @@ public async Task PollingClientShouldConnectQuickly(ClientAndServiceTestCase cli
 
         [Test]
         [LatestClientAndLatestServiceTestCases(testNetworkConditions: false, testListening: false)]
-        public async Task PollingClientShouldReConnectQuickly(ClientAndServiceTestCase clientAndServiceTestCase)
+        public async Task PollingClientShouldReConnectQuickly_WhenTheLastConnectionAttemptWasASuccess(ClientAndServiceTestCase clientAndServiceTestCase)
```
Comment (LukeButters) [inline source/Halibut.Tests/PollingClientConnectionHandlingFixture.cs:43]: I think this is the intent of the test, is that right @nathanwoctopusdeploy ?

File: source/Halibut.Tests/PollingClientConnectionHandlingFixture.cs [hunk 2]
```diff
@@ -101,6 +104,7 @@ public async Task PollingClientShouldReConnectQuickly(ClientAndServiceTestCase c
                 .WithPortForwarding()
                 .WithDoSomeActionService(doSomeActionServiceAction)
                 .WithEchoService()
+                .WithPollingReconnectRetryPolicy(() => new RetryPolicy(99999999, TimeSpan.Zero, TimeSpan.FromMinutes(1)))
```
Comment (LukeButters) [inline source/Halibut.Tests/PollingClientConnectionHandlingFixture.cs:107]: If we don't mark the last attempt as a success the time will be greater than zero resulting in a 1 min timeout.

File: source/Halibut/Util/RetryPolicy.cs [hunk 3]
```diff
@@ -10,7 +10,6 @@ public class RetryPolicy
         internal RetryPolicy(double backoffMultiplier, TimeSpan minimumDelay, TimeSpan maximumDelay)
         {
             if (backoffMultiplier <= 0) throw new ArgumentOutOfRangeException(nameof(backoffMultiplier), "Must be greater than zero");
-            if (backoffMultiplier > 1) throw new ArgumentOutOfRangeException(nameof(backoffMultiplier), "Must be less than or equal to one");
```
Comment (LukeButters) [inline source/Halibut/Util/RetryPolicy.cs:13]: Is this actually valuable?

**Review by nathanwoctopusdeploy — APPROVED** [review]
LGTM

**Pull request merged**